driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -264,8 +269,8 @@ private PoolingOptions options() {
         return leastBusy.setKeyspaceAsync(manager.poolsState.keyspace);
     }
 
-    private ListenableFuture<Connection> enqueue(int maxQueueSize) {
-        if (maxQueueSize == 0) {
+    private ListenableFuture<Connection> enqueue(long timeout, TimeUnit unit, int maxQueueSize) {
+        if (timeout == 0 || maxQueueSize == 0) {
```

Comment:
Nice side effect of this change!  Unless i'm misunderstanding, if we have to enqueue at all and timeout is 0 we move on to the next host immediately?